### DISCUSSION TITLE
General syntax discussion

### DISCUSSION THREAD

**Discussion by AliyevH** [MadeByMads/mad-migration#41] [General]
Hi. Lets' discuss syntax optimization.

**Comment by maestro-1** [comment]
Working on the async driver support currently and I'm almost done. So far I built using the sqlalchemy version 1.4 which is still in a pre-release stage. The alternatives are peewee-async, gino, tortoise, and ponyorm.
1. peewee-async is basically just an async wrapper around the original peewee orm, the way [databases](https://github.com/encode/databases) is an async wrapper for sqlalchemy and its usability(async) extends solely database queries
2. ponyorm maintenance is questionable.
3. gino would probably be the best in terms of alternatives but that would create a huge gap in the async and sync API of mad migration, since they are two completely independent orms. This is important to consider especially since async support for [sqlalchemy](https://docs.sqlalchemy.org/en/14/) is already available having a [beta](https://docs.sqlalchemy.org/en/14/) release (pre-release). 

Are we willing to go ahead with the prerelease of sqlalchemy until 1.4 is officially released or would it be preferable to make use of gino orm since it already had async support currently?

**Reply by Turall** [reply]
We used gino in production and everything is fine. In the Gino script, it mainly uses the sqlalchemy core to generate queries and execute them using asynchronous calls. However, madmigration does not currently handle any asynchronous calls, the logic is based on synchronous calls. We may have to create new objects to run asynchronously, which does not integrate with the current functionality. Sqlalchemy and Gino API are almost identical. We are very interested in how you implemented all this asynchronous, we would like to face to face talk about it.

**Reply by maestro-1** [reply]
You are absolutely right, the APIs for both of them are almost entirely the same, I'm guessing this has something to do with the fact that Gino was built on sqlalchemy with asynchronous use specifically in mind. I decided to go with sqlalchemy, I read the entirety of the async API and for the most part, it looks exactly like what was available before, so I decided to make use of that instead of gino.
